lib/backend/backend.go

```diff
@@ -437,6 +437,12 @@ const Separator = '/'
 // Key joins parts into path separated by Separator,
 // makes sure path always starts with Separator ("/")
 func Key(parts ...string) []byte {
+	return NewKey(parts...)
+}
+
+// NewKey joins parts into path separated by Separator,
+// makes sure path always starts with Separator ("/")
+func NewKey(parts ...string) []byte {
```

Comment:
Is `NewKey()` a valid invocation? Maybe `func NewKey(first string, rest ...string)` ?

In a similar line of thought, is it worth considering an erroring-variant, like `func NewKey(string, ...string) (*Key, error)` (and a `MustNewKey` to accompany it)?